File: LeetcodeJava/src/main/java/org/leetcode/algorithm/P3280.java
package org.leetcode.algorithm;

/**
 * Author: Deean
 * Date: 2025-03-09 10:39
 * FileName: src/main/java/org/leetcode/algorithm
 * Description:
 */

public class P3280 {
    class Solution {
        public String convertDateToBinary(String date) {
            String[] nums = date.split("-");
            for (int i = 0; i < nums.length; i++) {
                nums[i] = Integer.toBinaryString(Integer.parseInt(nums[i]));
            }
            return String.join("-", nums);
        }
    }

    public static void main(String[] args) {
        Solution s = new P3280().new Solution();
        Object ans = s.convertDateToBinary("2080-02-09");
        System.out.println(ans);
    }
}

File: LeetcodeCC/algorithm/P3280.cpp
/**
 * Author: Deean
 * Date: 2025-05-12 23:00
 * FileName: P3280.cpp
 * Description: 
 */

#include "../lib/leetcode.h"

using namespace std;

class Solution {
public:
    string binary(int x) {
        string s;
        while (x) {
            s.push_back('0' + (x & 1));
            x >>= 1;
        }
        reverse(s.begin(), s.end());
        return s;
    }

    string convertDateToBinary(string date) {
        int year = stoi(date.substr(0, 4));
        int month = stoi(date.substr(5, 2));
        int day = stoi(date.substr(8, 2));
        return binary(year) + "-" + binary(month) + "-" + binary(day);
    }
};

int main() {
    string ans = Solution().convertDateToBinary("2025-05-12");
    cout << ans << endl;
    return 0;
}
